src/time.hpp

```diff
@@ -24,4 +24,139 @@ inline double qdb_timespec_to_ms(const qdb_timespec_t & ts)
+    {
+        const int argc = 2;
+        auto seconds = static_cast<double>(ts.tv_sec);
+        auto nanoseconds = static_cast<double>(ts.tv_nsec);
+
+        v8::Local<v8::Value> argv[argc] = {
+            v8::Number::New(isolate, seconds),
+            v8::Number::New(isolate, nanoseconds),
+        };
+
+        v8::Local<v8::Function> cons = v8::Local<v8::Function>::New(isolate, constructor);
+
+        return (cons->NewInstance(isolate->GetCurrentContext(), argc, argv)).ToLocalChecked();
+    }
+
+    static bool InstanceOf(v8::Isolate * isolate, v8::Local<v8::Object> object)
+    {
+        auto ts_tmpl = Timestamp::tmpl.Get(isolate);
+        return ts_tmpl->HasInstance(object);
+    }
+
+    qdb_timespec_t getTimespec()
+    {
+        qdb_timespec_t ts;
+        ts.tv_sec = static_cast<qdb_time_t>(seconds_);
+        ts.tv_nsec = static_cast<qdb_time_t>(nanoseconds_);
+        return ts;
+    }
+
+private:
+    explicit Timestamp(double seconds, double nanoseconds) : seconds_(seconds), nanoseconds_(nanoseconds)
+    {
+    }
+
+    ~Timestamp()
+    {
+    }
+
+    static void New(const v8::FunctionCallbackInfo<v8::Value> & args)
+    {
+        auto isolate = args.GetIsolate();
+        if (args.IsConstructCall())
+        {
+            auto seconds = args[0]->IsUndefined() ? 0 : args[0]->NumberValue();
+            auto nanoseconds = args[1]->IsUndefined() ? 0 : args[1]->NumberValue();
+            auto obj = new Timestamp(seconds, nanoseconds);
+            obj->Wrap(args.This());
+            args.GetReturnValue().Set(args.This());
+        }
+        else
+        {
+            const int argc = 2;
+            v8::Local<v8::Value> argv[argc] = {args[0], args[1]};
+            auto cons = v8::Local<v8::Function>::New(isolate, constructor);
+            args.GetReturnValue().Set(cons->NewInstance(isolate->GetCurrentContext(), argc, argv).ToLocalChecked());
+        }
+    }
+
+    static void GetProperty(v8::Local<v8::String> property, const v8::PropertyCallbackInfo<v8::Value> & info)
+    {
+        auto isolate = info.GetIsolate();
+        auto obj = ObjectWrap::Unwrap<Timestamp>(info.This());
+
+        v8::String::Utf8Value s(property);
+        std::string str(*s, s.length());
+
+        if (str == "seconds")
+        {
+            info.GetReturnValue().Set(v8::Number::New(isolate, obj->seconds_));
+        }
+        else if (str == "nanoseconds")
+        {
+            info.GetReturnValue().Set(v8::Number::New(isolate, static_cast<double>(obj->nanoseconds_)));
+        }
+    }
+
+    static void FromDate(const v8::FunctionCallbackInfo<v8::Value> & args)
+    {
+        auto isolate = args.GetIsolate();
+
+        if (args.Length() != 1 || !args[0]->IsDate())
+        {
+            isolate->ThrowException(v8::Exception::TypeError(v8::String::NewFromUtf8(isolate, "Expected a Date")));
+        }
+
+        auto ms = args[0]->ToNumber(isolate)->Value();
```

Comment:
You should use the std::chrono library for this:
```
std::chrono::milliseconds ms = args[0]->ToNumber(isolate)->Value();
std::chrono::seconds seconds = ms; // will auto convert and truncate
std::chrono::nanoseconds nanoseconds = seconds - ms;
```
You can then access the number with `seconds.count()`

Reply:
`std::chrono::seconds seconds = ms; // will auto convert and truncate` Hmmm?
You need `duration_cast` here, since conversion from ms to s loses precision.